lib/plugins/aws/deploy/lib/extendedValidate.js

```diff
@@ -73,7 +72,7 @@ module.exports = {
           artifactFileName = this.provider.naming.getFunctionArtifactName(functionName);
           artifactFilePath = path.join(this.packagePath, artifactFileName);
 
-          if (_.get(functionObject, 'package.artifact')) {
```

Comment:
Are you sure it's safe? Is `package` guranteed to be set on `functionObject`?

Reply:
Looking at the service class, the package is not guaranteed to be set. Therefore, there needs to be a check if package has a value other than null/undefined.